### PR TITLE
Show recipe for cooking domain layer

### PR DESCRIPTION
**Pull request by arnekuchenbecker**

### CODE REVIEW

**Comment by kittyahy** [comment]
kotlin doc broke it.
(  Error:
OK: killing emulator, bye bye 
 OK)

File: app/src/main/java/com/scouts/kitchenplaner/model/usecases/DisplayRecipeForCooking.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                    }
+                    .map { name -> Pair(name, DietaryTypes.TRACE) }
+
+                val relevantFrees = it.freeOfAllergen
+                    .filter { free ->
+                        allergens.any { speciality ->
+                            speciality.allergen == free && speciality.type != DietaryTypes.FREE_OF
+                        }
+                    }
+                    .map { name -> Pair(name, DietaryTypes.FREE_OF) }
```
Comment (kittyahy) [inline app/src/main/java/com/scouts/kitchenplaner/model/usecases/DisplayRecipeForCooking.kt:83]: What if there the alternative improves on some ingredients on the alternative but is worse on another ingredient that the recipe to display is free of?

Line 86: all

**Review by kittyahy — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/scouts/kitchenplaner/model/usecases/DisplayRecipeForCooking.kt (hunk 1 above)
Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/model/usecases/DisplayRecipeForCooking.kt:83]: I dont understand the question

Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/model/usecases/DisplayRecipeForCooking.kt:83]: And no, line 86 should not be all, an allergen is not only relevant if all allergens of the main recipe have the same name as it.

**Review by kittyahy — APPROVED** [review]

**Review by kittyahy — APPROVED** [review]

**Pull request merged**